[#3886] Add Jackson module for Spring Data Page deserialization

theoema opened the pull request:
Adds a custom Jackson module that enables deserialization of Spring Data's
  Page interface into PageImpl instances. This resolves issues when using the
  JacksonConverter with paginated query results.

  The module is automatically registered as a Spring bean and picked up by
  Spring Boot's JacksonAutoConfiguration, which registers it with the ObjectMapper.
  The module only activates when Jackson is configured as the converter and
  Spring Data's Page interface is on the classpath.

  My hope is that this feels like a natural extension to the current Spring autoconfiguration,
  making it easy to implement more custom Jackson Modules in the future if needed.

  Implementation notes:
  - Module bean is conditionally registered based on converter configuration
  - Deserializer handles standard Page JSON structure (content, number, size, totalElements)
  - Falls back to sensible defaults when fields are missing
  - Registered in AutoConfiguration.imports for automatic discovery

theoema commented on extensions/spring/spring-boot-autoconfigure/src/main/java/org/axonframework/extension/springboot/autoconfig/CustomJacksonModuleAutoConfiguration.java:
```diff
@@ -0,0 +1,147 @@
+     *
+     * @see CustomJacksonModuleAutoConfiguration#springDataPageJacksonModule()
+     */
+    private static class JacksonPageDeserializer extends JsonDeserializer<Page<?>> {
```
Let me take a look at both points and get back to you! @hjohn @smcvb

theoema replied:
@smcvb @hjohn Hey guys. I did some digging and playing around and heres my findings:

No, Spring (Data) does not provide a JsonDeserializer for Page, I guess they never really thought it would be needed, since its just meant to be a return type. So sadly we have to define and maintain our own custom deserializer. On a brighter side the Spring Data Page interface has remained largely unchanged for the last decade and a half, so this shouldn't be a big maintainence burden. 

As for your suggestion @hjohn , I did play around with the abstract type mappings for both Page and the nested Pageable, trying to get it to work, but I didn't have any success, perhaps if we dug deeper into the ObjectMapper logic for Messages we may be able to get it to work, but I would assume that you would like to keep a "spring specific" piece of configuration out of the main "framework agnostic" code, so I think the solution I provided here is close to optimal.

If you have any other thoughts or considerations, please share them, and have a good weekend guys!

smcvb commented on extensions/spring/spring-boot-autoconfigure/src/main/java/org/axonframework/extension/springboot/autoconfig/CustomJacksonModuleAutoConfiguration.java:
```diff
@@ -0,0 +1,147 @@
+    @Bean
+    @ConditionalOnClass(Page.class)
+    @Conditional(JacksonConfiguredCondition.class)
+    public Module springDataPageJacksonModule() {
```
I think it's nicer if we group this bean creation method in the `ObjectMapperAutoConfiguration`, as it's `ObjectMapper` details that we are introducing.

theoema commented:
## Refactoring Complete

Hi @smcvb, I've addressed your review feedback with the following changes:

### What changed:

1. **Moved `JacksonPageDeserializer` to axon-spring module**
   - New location: `org.axonframework.extension.spring.data.JacksonPageDeserializer`
   - As you noted, this is Spring Data specific, not Spring Boot specific

2. **Moved bean creation to `ObjectMapperAutoConfiguration`**
   - The `springDataPageJacksonModule()` bean is now in `ObjectMapperAutoConfiguration.java`
   - Grouped with other ObjectMapper-related configuration

3. **Added Spring Data BOM to parent pom**
   - Added `spring-data.version` property (2025.0.7, aligned with Spring Boot 3.5.9)
   - Added `spring-data-bom` import following the same pattern as `spring-framework-bom`
   - Added `spring-data-commons` as optional dependency in axon-spring module

4. **Split test file**
   - Deserializer unit tests → `JacksonPageDeserializerTest` in axon-spring
   - Bean registration tests → `ObjectMapperAutoConfigurationTest` in spring-boot-autoconfigure

5. **Removed `CustomJacksonModuleAutoConfiguration`**
   - Class deleted
   - Entry removed from `AutoConfiguration.imports`

Let me know if you have any further feedback!
